doc/content/en/docs/next/Specification/_index.md

```diff
@@ -251,7 +250,9 @@ The fullname of a record, enum or fixed definition is determined by the required
 
 References to previously defined names are as in the latter two cases above: if they contain a dot they are a fullname, if they do not contain a dot, the namespace is the namespace of the enclosing definition.
 
-Primitive type names have no namespace and their names may not be defined in any namespace.
+Primitive type names (`null`, `boolean`, `int`, `long`, `float`, `double`, `bytes`, `string`) have no namespace and their names may not be defined in any namespace.
+
+Complex types (`record`, `enum`, `array`, `map`, `fixed`) have no namespace, but their names (as well as `union`) are permitted to be reused as type names.  This can be confusing to the human reader, but is always unambiguous for binary serialization.  Due to the limitations of JSON encoding, it is a best practice to use a namespace with these names.
```

Comment:
Initially, I read this as "it is best to use these names in a namespace"...

Can we use phrasing like this instead?
"..., it is a best practice to use a namespace when using these names."